src/buttons/Button.tsx

```diff
@@ -52,174 +52,167 @@ export type ButtonProps = TouchableOpacityProps &
     raised?: boolean;
     theme?: Theme;
   };
-
-class Button extends Component<ButtonProps, {}> {
-  componentDidMount() {
-    const { linearGradientProps, ViewComponent } = this.props;
+const Button: React.FunctionComponent<ButtonProps> = (props) => {
+  useMemo(() => {
+    const { linearGradientProps, ViewComponent } = props;
     if (linearGradientProps && !ViewComponent) {
       console.error(
         "You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}"
       );
     }
-  }
+  }, [props]);
```

Comment:
You have `useMemo` here, but it triggers on all changes to props? I don't think this is a better solution.